libsplinter/src/biome/users/mod.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod database;
+mod error;
+pub(in crate::biome) mod user_store;
+
+pub(in crate::biome) use error::UserStoreError;
+
+use database::models::UserModel;
+
+pub struct SplinterUser {
+    id: String,
+}
+
+impl SplinterUser {
+    pub fn new(user_id: &str) -> Self {
+        SplinterUser {
+            id: user_id.to_string(),
+        }
+    }
+
+    pub fn id(&self) -> &str {
+        &self.id
+    }
+}
+
+/// Defines methods for CRUD operations and fetching and listing users
+/// without defining a storage strategy
+pub trait UserStore<T> {
+    /// Adds a user to the underlying storage
+    ///
+    /// # Arguments
+    ///
+    ///  * `user` - The user to be added
+    ///
+    ///
+    fn add_user(&self, user: T) -> Result<(), UserStoreError>;
+
+    /// Updates a user information in the underling storage
+    ///
+    /// # Arguments
+    ///
+    ///  * `user` - The user with the updated information
+    ///
+    fn update_user(&self, updated_user: T) -> Result<(), UserStoreError>;
+
+    /// Removes a user from the underlying storage
+    ///
+    /// # Arguments
+    ///
+    ///  * `identifier` - The unique identifier of the user to be removed
+    ///
+    fn remove_user(&self, identifier: String) -> Result<T, UserStoreError>;
```

Comment:
What field does identifier map to? is it the user_id?

Reply:
updated to say id